utils/utils.go

```diff
@@ -51,32 +53,67 @@ func eofKeySequenceText() string {
 }
 
 // AddSecret Recalls state if present, and appends a secret to the state file
-func AddSecret(toAdd Secret) {
-	ism := NewIOStateManager(defaultFile)
-	secrets, _ := ism.ReadState()
+func AddSecret(toAdd Secret, keyURI string, keyCheck bool) {
+	envelope := V2{Filepath: defaultFile}
+	envelope.KeyIdentifier = keyURI
+	nvl := NewVersionedLoader(defaultFile)
+
+	contents, err := nvl.ReadState()
+	// First run case with FileNotFound exception. Mask this and save
+	if os.IsNotExist(err) {
+		envelope.Secrets.Add(toAdd)
+		envelope.Save()
+		return
+	}
```

Comment:
This does not check for any other error state, which could lead to other issues being silenced.

```go
contents, err := nvl.ReadState()
if err != nil {
	if os.IsNotExist(err) {
		...
	} else {
		...
	}
}
```

Reply:
oh, nevermind.. I see there is a check below which would check for an error here. I think that it should at least be moved up since there is no other error declaration and separating related code blocks can make things difficult to follow